src/routes/router.js

```diff
@@ -1,9 +1,35 @@
+const URL = require("url");
+const querystring = require("querystring");
+
 const mainRoute = require("./main/main");
-const productsRoute = require("./products/productsRoute");
+const productsRoute = require("./products/allProducts");
+const idProductRoute = require("./products/id");
+const allIdProducts = require("./products/allIds");
 const signupRoute = require("./signup/signupReoute");
-const router = {
-  "/products": productsRoute,
-  "/signup": signupRoute,
-  default: mainRoute
+const productsByCategory = require("./products/categories");
+
+const router = url => {
+  const reqUrl = URL.parse(url);
+  const queryStr = querystring.parse(reqUrl.query);
+
+  const productId = url.slice(10);
+
+  if (queryStr.ids) {
+    return allIdProducts;
+  }
+  if (queryStr.category) {
+    return productsByCategory;
+  }
+  if (url === "/products") {
+    return productsRoute;
+  }
+  if (url === `/products/${productId}`) {
+    return idProductRoute;
+  }
+  if (url === "/signup") {
+    return signupRoute;
+  }
+  return mainRoute;
 };

```

Comment:
is it better to rewrite in to the object?

Reply:
yes, rewrite to object